app/uk/gov/hmrc/incometaxsessiondata/services/SessionService.scala

```diff
@@ -27,18 +27,25 @@ class SessionService @Inject() (
   repository: SessionDataRepository
 )(implicit ec: ExecutionContext) {
 
-  def get(sessionId: String): Future[Either[Throwable, Option[SessionData]]] =
-    repository.get(sessionId) map {
+  def get(request: SessionDataRequest[_]): Future[Either[Throwable, Option[SessionData]]] =
+    repository.get(request) map {
       case Some(data: Session) =>
-        Right(Some(data))
+        Right(Some(SessionData.fromSession(data)))
       case None                => Right(None)
     }
 
-  def set(sessionData: SessionData): Future[Boolean] =
-    repository.set(sessionData)
+  def getByMtditid(mtditid: String): Future[Seq[Session]] =
+    repository.getByMtditid(mtditid)
 
-  def deleteSession(sessionId: String): Future[Unit] =
-    repository.deleteOne(sessionId).map {
+  def set(session: Session): Future[Either[Throwable, Boolean]] =
+    try
```

Comment:
if we can avoid use of try/catch block in favour of [Try](https://www.scala-lang.org/api/2.13.6/scala/util/Try.html)